---
description: 编译器错误 CS0155
title: 编译器错误 CS0155
ms.date: 07/20/2015
f1_keywords:
- CS0155
helpviewer_keywords:
- CS0155
ms.assetid: 6c92984a-2b10-453e-9cb7-e6a1d1b98aa6
ms.openlocfilehash: fb284e83f2cf4078fe5180a74adf76f7e9948b6f
ms.sourcegitcommit: d579fb5e4b46745fd0f1f8874c94c6469ce58604
ms.translationtype: MT
ms.contentlocale: zh-CN
ms.lasthandoff: 08/30/2020
ms.locfileid: "89121518"
---
# <a name="compiler-error-cs0155"></a><span data-ttu-id="e776f-103">编译器错误 CS0155</span><span class="sxs-lookup"><span data-stu-id="e776f-103">Compiler Error CS0155</span></span>
<span data-ttu-id="e776f-104">捕获或抛弃的类型必须从 System.Exception 派生</span><span class="sxs-lookup"><span data-stu-id="e776f-104">The type caught or thrown must be derived from System.Exception</span></span>  
  
 <span data-ttu-id="e776f-105">尝试将不是派生自 **System.Exception** 的数据类型传递到 [catch](../language-reference/keywords/try-catch.md) 块中。</span><span class="sxs-lookup"><span data-stu-id="e776f-105">An attempt was made to pass a data type that does not derive from **System.Exception** into a [catch](../language-reference/keywords/try-catch.md) block.</span></span> <span data-ttu-id="e776f-106">只有派生自 **System.Exception** 的数据类型才能传递到 **catch** 块中。</span><span class="sxs-lookup"><span data-stu-id="e776f-106">Only data types that derive from **System.Exception** can be passed into a **catch** block.</span></span> <span data-ttu-id="e776f-107">有关详细信息，请参阅 [异常和异常处理](../programming-guide/exceptions/index.md)。</span><span class="sxs-lookup"><span data-stu-id="e776f-107">For more information, see [Exceptions and Exception Handling](../programming-guide/exceptions/index.md).</span></span>  
  
 <span data-ttu-id="e776f-108">下面的示例生成 CS0155：</span><span class="sxs-lookup"><span data-stu-id="e776f-108">The following sample generates CS0155:</span></span>  
  
```csharp  
// CS0155.cs  
using System;  
  
namespace MyNamespace  
{  
    public class MyClass2  
    // try the following line instead  
    // public class MyClass2 : Exception  
    {  
    }  
    public class MyClass  
    {  
        public static void Main()  
        {  
            try  
            {  
            }  
            catch (MyClass2)   // CS0155, resolves if you derive MyClass2 from Exception  
            {  
            }  
        }  
    }  
}  
```
